pkg/apis/cassandraoperator/v1alpha1/cassandradatacenter.go

```diff
@@ -10,18 +10,20 @@ import (
 type CassandraDataCenterSpec struct {
 	// Cluster is either a string or v1.LocalObjectReference
 	//Cluster interface{} `json:"cluster,omitempty"`
-	Cluster          string                    `json:"cluster,omitempty"`
-	Nodes            int32                     `json:"nodes"`
-	CassandraImage   string                    `json:"cassandraImage"`
-	SidecarImage     string                    `json:"sidecarImage"`
-	ImagePullPolicy  v1.PullPolicy             `json:"imagePullPolicy"`
-	ImagePullSecrets []v1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
-
-	Resources v1.ResourceRequirements `json:"resources"`
-
-	DataVolumeClaimSpec v1.PersistentVolumeClaimSpec `json:"dataVolumeClaimSpec"`
-
-	PrometheusSupport bool `json:"prometheusSupport"`
+	Cluster                   string                       `json:"cluster,omitempty"`
+	Nodes                     int32                        `json:"nodes"`
+	CassandraImage            string                       `json:"cassandraImage"`
+	SidecarImage              string                       `json:"sidecarImage"`
+	ImagePullPolicy           v1.PullPolicy                `json:"imagePullPolicy"`
+	ImagePullSecrets          []v1.LocalObjectReference    `json:"imagePullSecrets,omitempty"`
+	BackupSecretVolume        v1.SecretVolumeSource        `json:"backupSecretVolume,omitempty"`
```

Comment:
`BackupSecretVolume` and `UserSecretVolume` should be named `...VolumeSource` for consistency with `UserConfigMapVolumeSource`.

If these fields are optional, they should be pointers.

Reply:
yes, I've thought about it after pushing, and was already too sleepy to think about it too much, so sure, will rename.